File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260921-202808-59615/analysis/acceptance-project/src/main/java/surveycrawler/CrawlerRun.java
package surveycrawler;

public abstract class CrawlerRun {

    // guarded run

    public abstract void startGuardedRun(SurveyCrawler aCrawler);

    public abstract void endGuardedRun(SurveyCrawler aCrawler);

    public abstract void assertIsFinished(SurveyCrawler aCrawler);

    // moving

    public abstract void slideOnSilt(SurveyCrawler aCrawler, Point aDirection, SiltSlide aSlide);

    // undoing

    public abstract void registerUndoAction(Runnable anUndoAction);

    public abstract void handleErrorOf(SurveyCrawler aCrawler);
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-202806-59410/analysis/own-tests-project/src/surveycrawler/crawler_run.py
class CrawlerRun:

    # guarded run

    def start_guarded_run(self, a_crawler):
        raise NotImplementedError()

    def end_guarded_run(self, a_crawler):
        raise NotImplementedError()

    def finish(self, a_crawler):
        raise NotImplementedError()

    # undoing

    def register_undo(self, an_undo_action):
        raise NotImplementedError()

    def handle_error(self, a_crawler):
        raise NotImplementedError()

    # moving

    def slide_crawler_towards(self, a_crawler, a_direction, a_silt):
        raise NotImplementedError()
